### PR TITLE
feat(executor): upgrade Cairo compiler to 2.4.0-rc2 and blockifier to 0.4.0-rc6

### LINKED ISSUE #1482: Upgrade blockifier to v0.13 compatible version
Such a version does not yet exist, but should be shortly. This will allow us to execute the new V3 transactions.

### LINKED ISSUE #1520: Upgrade sierra compiler
This will be backwards compatible and should be a drop-in replacement for our latest current compiler.

### PR DESCRIPTION
**Pull request by kkovaacs**
This PR upgrades our execution related crates to the versions required by Starknet 0.13.0.

A new `starknet_api` crate version is required so that we can represent v3 transactions and some blockifier APIs have also changed.

Closes #1520 
Closes #1482

### CODE REVIEW

File: crates/executor/src/block_context.rs [hunk 1]
```diff
@@ -39,9 +39,16 @@ pub(super) fn construct_block_context(
             PatriciaKey::try_from(execution_state.header.sequencer_address.0.into_starkfelt())
                 .expect("Sequencer address overflow"),
         ),
-        fee_token_address,
+        fee_token_addresses: blockifier::block_context::FeeTokenAddresses {
+            // FIXME: what is the STRK token address?
+            strk_fee_token_address: fee_token_address,
```
Comment (Mirko-von-Leipzig) [inline crates/executor/src/block_context.rs:44]: We should find out :)

File: crates/rpc/src/executor.rs [hunk 2]
```diff
@@ -189,16 +186,14 @@ pub(crate) fn map_broadcasted_transaction(
...
-                    ),
-                    nonce: starknet_api::core::Nonce(Felt::ZERO.into_starkfelt()),
                 };
 
                 let tx = pathfinder_executor::Transaction::from_api(
                     starknet_api::transaction::Transaction::Invoke(
                         starknet_api::transaction::InvokeTransaction::V0(tx),
                     ),
+                    starknet_api::transaction::TransactionHash(transaction_hash.0.into_starkfelt()),
+                    None,
```
Comment (Mirko-von-Leipzig) [inline crates/rpc/src/executor.rs:196]: What does this new `None` field do?

**Review by Mirko-von-Leipzig — APPROVED** [review]
LGTM

File: crates/rpc/src/executor.rs (hunk 2 above)
Comment (kkovaacs) [reply crates/rpc/src/executor.rs:196]: That's `deployed_contract_address` -- we have it only for DEPLOY_ACCOUNT transactions.

**Pull request merged**